client/src/components/UnapprovedPosts.jsx

```diff
@@ -0,0 +1,62 @@
+import { useEffect, useState } from "react";
+import "./PostView.css";
+import { approvePost, getUnapprovedPosts } from "../managers/postManager";
+
+const UnapprovedPosts = () => {
+   
+
+    const [unapprovedPosts, setUnapprovedPosts] = useState([]);
+
+    const getAndSetUnapprovedPosts = () => {
+        getUnapprovedPosts().then(setUnapprovedPosts)
+    }
+
+    useEffect(() => {
+        getAndSetUnapprovedPosts();
+    }, []);
+
+    const handleApproveClick = (event) => {
+        approvePost(event.target.value).then(getAndSetUnapprovedPosts())
```

Comment:
i think you meant to pass the `getAndSetUnapprovedPosts` function to the `.then`. Calling the function within the `.then` will pass the return of the function, which is nothing. In other words, you'll need to remove the parenthesis in `getAndSetUnapprovedPosts()`.

Reply:
ty!